### DISCUSSION TITLE
How to use visitors

### DISCUSSION THREAD

**Discussion by victorinno** [rrevenantt/antlr4rust#28] [Q&A]
Hi there. I would like to ask how can visitors be used to write values.

For example I have an INT: [1-9]; and another SUM: INT '+' INT.
In vist_int how can I convert the text to i64 for example and how can I get it the results in visit_sum?

**Comment by pm100** [comment]
@victorinno I asm sure you are long gone but here is my visit_number 

```
    fn visit_number(&mut self, ctx: &NumberContext) -> TermValue {
        println!("visit_number");
        let text = ctx.get_text();
        TermValue::Number(text.parse().unwrap())
    }
```
